sycl/plugins/esimd_cpu/pi_esimd_cpu.cpp

```diff
@@ -683,13 +1097,48 @@ pi_result piextQueueCreateWithNativeHandle(pi_native_handle, pi_context,
 pi_result piMemBufferCreate(pi_context Context, pi_mem_flags Flags, size_t Size,
                             void *HostPtr, pi_mem *RetMem,
                             const pi_mem_properties *properties) {
-  (void)Context;
-  (void)Flags;
-  (void)Size;
-  (void)HostPtr;
-  (void)RetMem;
-  (void)properties;
-  DIE_NO_IMPLEMENTATION;
+  if ((Flags & PI_MEM_FLAGS_ACCESS_RW) == 0) {
+    return PI_INVALID_OPERATION;
+  }
```

Comment:
Could you please clarify why this operation is considered to be invalid?

Reply:
As far as I know, CM does not support buffer for READONLY or WRITEONLY.

Reply:
FIXME: lots of tests will crash, as read-only or write-only accessor creation requests are routine. You should instead map R, W, RW to RW then.

Reply:
The access mode passed to this function doesn't depend on the accessor mode. As of now SYCL RT should always create buffers in RW mode.

Reply:
> The access mode passed to this function doesn't depend on the accessor mode. As of now SYCL RT should always create buffers in RW mode.

If so, I'll remove this sanity check as 'readonly/writeonly' atrribute won' be used at PI level.

Reply:
Please leave the sanity check, usages of `readonly/writeonly` can be added in future.
Actually it's even added recently for very specific case: https://github.com/intel/llvm/pull/4015/files?file-filters%5B%5D=.cpp&file-filters%5B%5D=.def&file-filters%5B%5D=.dump&file-filters%5B%5D=.h&file-filters%5B%5D=.hpp#diff-5203875534d6d4be5e5b7a8608fb09a6be772176f24da0ec74ed7729a9e6dc8aR132

Reply:
Yes, I'll leave the check. I noticed that after I put my comment.